### DISCUSSION TITLE
java.lang.NoClassDefFoundError software/amazon/awssdk/services/s3/crt/S3CrtSdkHttpExecutionAttribute

### DISCUSSION THREAD

**Discussion by ivanmgb** [aws/aws-sdk-java-v2#4704] [Q&A]
### Describe the bug

I downloaded the S3 Transfer manager dependency on Maven repo
`<dependency>
  <groupId>software.amazon.awssdk</groupId>
<artifactId>s3-transfer-manager</artifactId>
    <version>2.21.22</version>
</dependency>`

Setup my configuration and run my app and I got this error.
Running mvn dependency:tree, I noticed that one of the transitive dependency is `software.amazon.awssdk:s3:jar:2.20.63`

![image](https://github.com/aws/aws-sdk-java-v2/assets/86414233/bfe3bc82-5207-487b-87df-b7529949ba83)

Instead of pulling version 2.21.22 as described under the compille dependencies section: https://mvnrepository.com/artifact/software.amazon.awssdk/s3-transfer-manager/2.21.22

The class S3CrtSdkHttpExecutionAttribute was introduced in https://javadoc.io/static/software.amazon.awssdk/s3/2.21.22/software/amazon/awssdk/services/s3/crt/package-summary.html 

and is not in version 2.20.63
https://javadoc.io/static/software.amazon.awssdk/s3/2.20.63/software/amazon/awssdk/services/s3/crt/package-summary.html

### Expected Behavior

It should pull the correct version based on the dependencies reported by Maven

### Current Behavior

The dependency s3-transfer-manager version 2.21.22 pulled software.amazon.awssdk:s3 version 2.20.63

### Reproduction Steps

Added the following in the pom.xml

       <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>s3-transfer-manager</artifactId>
            <version>2.21.22</version>
        </dependency>

        <dependency>
            <groupId>software.amazon.awssdk.crt</groupId>
            <artifactId>aws-crt</artifactId>
            <version>0.28.0</version>
        </dependency>

Then run mvn dependency:tree

### AWS Java SDK version used

V2

### JDK version used

OpenJDK Runtime Environment Zulu17.44+53-CA (build 17.0.8.1+1-LTS)

### Operating System and version

Mac Ventura 13.5.2

**Comment by debora-ito** [comment]
@ivanmgb can you share your full pom.xml?
My guess is that some other dependency in your pom is pulling in s3 2.20.63

I'll move this to Github Discussion, since it's a guidance question.

**Reply by ivanmgb** [reply]
Hi @debora-ito Here is the full dependency list of my pom

> <dependencies>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
        </dependency>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-text</artifactId>
            <version>1.11.0</version>
        </dependency>
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>s3-transfer-manager</artifactId>
            <version>2.20.63</version>
        </dependency>
        <dependency>
            <groupId>software.amazon.awssdk.crt</groupId>
            <artifactId>aws-crt</artifactId>
            <version>0.21.12</version>
        </dependency>
    </dependencies>

I'll check if there are conflicting dependencies. But currently my workaround is to use an older version of transfer manager and aws-crt. Thank you

**Reply by debora-ito** [reply]
To clarify: you have `software.amazon.awssdk:s3-transfer-manager` declared in two places in the pom file, with versions `2.20.63` and `2.21.22`? Because the versions in this pom snippet are different from the pom in the issue description.

Can you share the full pom, not just the dependency list?
